charts/opentelemetry-collector/templates/_config.tpl

```diff
@@ -53,6 +63,9 @@ Build config file for deployment OpenTelemetry Collector
 {{- $values := deepCopy .Values }}
 {{- $data := dict "Values" $values | mustMergeOverwrite (deepCopy .) }}
 {{- $config := include "opentelemetry-collector.baseConfig" $data | fromYaml }}
+{{- if .Values.presets.hostMetrics.enabled }}
```

Comment:
Should we allow hostmetrics to be used in deployment mode or keep it to daemonset mode like filelog is?

Reply:
I don't think we should care about such misconfigurations. Probably just mention that `hostMetrics` supposed to be used with daemonset deployment only

Reply:
I would agree.  Kind feels like we should remove the restriction from logsCollection as well.

Reply:
> Should we allow hostmetrics to be used in deployment mode or keep it to daemonset mode like filelog is?